datamap/bloom/src/main/java/org/apache/carbondata/datamap/bloom/BloomDataMapCache.java

```diff
@@ -0,0 +1,197 @@
+    public static final BloomDataMapCache INSTANCE = new BloomDataMapCache();
+  }
+
+  /**
+   * get instance
+   */
+  public static BloomDataMapCache getInstance() {
+    return SingletonHolder.INSTANCE;
+  }
+
+  /**
+   * for resolve from serialized
+   */
+  protected Object readResolve() {
+    return getInstance();
+  }
+
+  /**
+   * load datamap from bloomindex file
+   */
+  private List<BloomDMModel> loadBloomDataMapModel(CacheKey cacheKey) {
+    DataInputStream dataInStream = null;
+    ObjectInputStream objectInStream = null;
+    List<BloomDMModel> bloomDMModels = new ArrayList<BloomDMModel>();
+    try {
+      String indexFile = getIndexFileFromCacheKey(cacheKey);
+      dataInStream = FileFactory.getDataInputStream(indexFile, FileFactory.getFileType(indexFile));
+      objectInStream = new ObjectInputStream(dataInStream);
+      try {
+        BloomDMModel model = null;
+        while ((model = (BloomDMModel) objectInStream.readObject()) != null) {
+          bloomDMModels.add(model);
+        }
+      } catch (EOFException e) {
+        LOGGER.info(String.format("Read %d bloom indices from %s",
+            bloomDMModels.size(), indexFile));
+      }
+      this.bloomDMCache.put(cacheKey, bloomDMModels);
+      return bloomDMModels;
+    } catch (ClassNotFoundException | IOException e) {
+      LOGGER.error(e, "Error occurs while reading bloom index");
+      throw new RuntimeException("Error occurs while reading bloom index", e);
+    } finally {
+      clear();
+      CarbonUtil.closeStreams(objectInStream, dataInStream);
+    }
+  }
+
+  /**
+   * get bloom index file name from cachekey
+   */
+  private String getIndexFileFromCacheKey(CacheKey cacheKey) {
+    return cacheKey.shardPath.concat(File.separator).concat(cacheKey.indexColumn)
+        .concat(BloomCoarseGrainDataMap.BLOOM_INDEX_SUFFIX);
+  }
+
+  /**
+   * get bloom datamap from cache
+   */
+  public List<BloomDMModel> getBloomDMModelByKey(CacheKey cacheKey) {
+    return this.bloomDMCache.getUnchecked(cacheKey);
+  }
+
+  /**
+   * get cache status
+   */
+  private String getCacheStatus() {
+    StringBuilder sb = new StringBuilder();
+    CacheStats stats = this.bloomDMCache.stats();
+    sb.append("hitCount: ").append(stats.hitCount()).append(System.lineSeparator())
+        .append("hitRate: ").append(stats.hitCount()).append(System.lineSeparator())
+        .append("loadCount: ").append(stats.loadCount()).append(System.lineSeparator())
+        .append("averageLoadPenalty: ").append(stats.averageLoadPenalty())
+        .append(System.lineSeparator())
+        .append("evictionCount: ").append(stats.evictionCount());
+    return sb.toString();
+  }
+
+  /**
+   * clear this cache
+   */
+  private void clear() {
+    LOGGER.error(String.format("Current meta cache statistic: %s", getCacheStatus()));
+    LOGGER.error("Trigger invalid all the cache for bloom datamap");
+    this.bloomDMCache.invalidateAll();
+  }
+
+  public static class CacheKey {
```

Comment:
public is not required

Reply:
No, in BloomDataMap class we need to create a CacheKey instance to get value from cache.
However I found another place similar with this and fixed it